File: LeetCode/104-maximum-depth-of-binary-tree.py
#!/usr/bin/env python
#coding=utf-8

# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution(object):
    def maxDepth(self, root):
        """
        :type root: TreeNode
        :rtype: int
        """
        return self.dfs(root)
        
    def dfs(self, root):
        if(root == None):
            return 0
        return 1 + max(self.dfs(root.left), self.dfs(root.right))

File: LeetCode/104-maximum-depth-of-binary-tree.java
public class Solution {
    public int maxDepth(TreeNode root) {
        return dfs(root);
    }
    private int dfs(TreeNode root){
        if(root == null) return 0;
        return 1 + Math.max(dfs(root.left), dfs(root.right));
    }
}
